src/main/java/org/spongepowered/common/entity/player/SpongeUser.java

```diff
@@ -183,79 +184,83 @@ public DataContainer toContainer() {
                 .set(DataQueries.USER_SPAWNS, this.spawnLocations);
     }
 
+    private NotImplementedException createNotImplementedException() {
+        return new NotImplementedException("User is offline, offline inventory API yet to be implemented");
+    }
+
     @Override
     public boolean canEquip(EquipmentType type) {
-        throw new UnsupportedOperationException(); // TODO Inventory API
+        return this.self.getPlayer().orElseThrow(this::createNotImplementedException).canEquip(type); // TODO Inventory API
```

Comment:
You can put the complete `this.self.getPlayer().orElseThrow(...)` part into an extra method (e.g. `getPlayerIfOnline()`, then you have even less to write for each method.